Lib/dataclasses.py

```diff
@@ -1015,6 +1015,11 @@ def _process_class(cls, init, repr, eq, order, unsafe_hash, frozen,
         # Does this class have a post-init function?
         has_post_init = hasattr(cls, _POST_INIT_NAME)
 
+        # typing.Protocol can override __init__ method to object.__init__
+        inherits_from_protocol = any(getattr(c, '_is_protocol', False) for c in cls.__bases__)
+        if inherits_from_protocol and cls.__dict__.get('__init__') is object.__init__:
+            del cls.__init__
+
```

Comment:
Could you improve this comment to more fully explain what's going on? I don't understand the issue by looking at this. What is it special about typing.Protocol that's different from any other class's implementation of __init__ that requires it to be special cased?

Reply:
Looks like I found a way how to fix this issue without changing `dataclasses` module.